tests/unit/remote/test_remote_dir.py

```diff
@@ -97,15 +92,39 @@ def test_copy_preserve_etag_across_buckets(remote):
 
     remote.copy(from_info, to_info)
 
-    from_etag = RemoteS3.get_etag(s3, "bucket", "foo")
+    from_etag = RemoteS3.get_etag(s3, from_info.bucket, from_info.path)
     to_etag = RemoteS3.get_etag(s3, "another", "foo")
 
     assert from_etag == to_etag
 
 
+@pytest.mark.parametrize("remote", remotes, indirect=True)
 def test_makedirs(remote):
     empty_dir = remote.path_info / "empty_dir" / ""
     remote.remove(empty_dir)
     assert not remote.exists(empty_dir)
     remote.makedirs(empty_dir)
     assert remote.exists(empty_dir)
+    assert remote.isdir(empty_dir)
+
+
+@pytest.mark.parametrize("remote", [GCP], indirect=True)
+def test_isfile(remote):
+    test_cases = [
+        (False, "empty_dir/"),
+        (True, "empty_file"),
+        (True, "foo"),
+        (True, "data/alice"),
+        (True, "data/alpha"),
+        (True, "data/subdir/1"),
+        (True, "data/subdir/2"),
+        (True, "data/subdir/3"),
+        (False, "data/subdir/empty_dir/"),
+        (True, "data/subdir/empty_file"),
+        (False, "something-that-does-not-exist"),
+        (False, "data/subdir/empty-file/"),
+        (False, "empty_dir"),
+    ]
+
+    for expected, path in test_cases:
+        assert remote.isfile(remote.path_info / path) == expected
```

Comment:
This test will need adjustment if #2873 gets merged.

Reply:
Yeah, let's merge #2873 first, once we get one more approval 😉 And then we'll merge this one.

Reply:
@skshetry  #2873 is merged, please adjust this PR accordingly 🙂